server/ladder_service.py

```diff
@@ -125,8 +131,22 @@ async def fetch_map_pools(self, conn) -> Dict[int, Tuple[str, List[Map]]]:
             _, map_list = map_pool_maps[id_]
             if row.map_id is not None:
                 map_list.append(
-                    Map(row.map_id, row.display_name, row.filename)
+                    Map(row.map_id, row.display_name, row.filename, row.weight)
                 )
+            elif row.other_map_params is not None:
+                param_dict = json.loads(row.other_map_params)
+                if "type" in param_dict:
+                    map_type = param_dict["type"]
+                    if map_type == "neroxis":
+                        if "size" in param_dict and "spawns" in param_dict and "version" in param_dict:
+                            map_size_pixels = param_dict["size"]
+                            spawns = param_dict["spawns"]
+                            version = param_dict["version"]
+                            if isinstance(map_size_pixels, int) and isinstance(spawns, int):
+                                gen_id = version + str(spawns) + str(map_size_pixels)
```

Comment:
Maybe override the constructor to do this automatically. Also a python tip, strings are immutable so it's better to format than add when you have more than 2 strings, otherwise it will lead to excessive copying:
```suggestion
                                gen_id = f"{version}{spawns}{map_size_pixels}"
```

Reply:
ah yeah that was just a derp moment of me forgetting about f strings

Reply:
And unfortunately you can't overwrite the __init__ of NamedTuples

Reply:
I guess you could make a class method and just have it call the constructor :P.